kcauto.sikuli/combat.sikuli/combat.py

```diff
@@ -404,14 +402,11 @@ def _run_combat_logic(self):
                     Util.click_preset_region(self.regions, 'center')
                     Util.rejigger_mouse(self.regions, 'lbas')
 
-                combat_result = self._run_loop_during_battle()
-
-                # resolve night battle
-                if combat_result == 'night_battle':
-                    if self._select_night_battle(self._resolve_night_battle()):
-                        self._run_loop_during_battle()
-
-                self.regions['lower_right_corner'].wait('next.png', 30)
+                while not self.regions['lower_right'].exists('next.png', 1 + Globals.SLEEP_MODIFIER):
```

Comment:
Ah, you're right. I had a brain fart and automatically started thinking about `Util.kc_sleep` since that's really only been the place where `Globals.SLEEP_MODIFIER` was used. I see what you're trying to do here. Let me think about this a bit.

Reply:
The more I think about it I feel that most of the `Globals.SLEEP_MODIFIER` modifiers on these `exists` calls are not necessary. The ones in `while` loops naturally resolve themselves if they're just set to 1second, The ones outside of `while` loops can just be blank, which defaults to 3seconds. I don't think people generally set a sleep modifier above 3 in the first place.

Reply:
Yes. `What You See is What You Script.` When we're playing a game, we're not just waiting for buttons - we're looking for them. The implementation makes the code very flexible, I can even set up a countdown timer and/or crash assets detection to call recovery within it. The only point that I see as an obstacle is possible CPU overconsumption. But I'd like to disregard it.
Another point is too fast clicking. In theory `Globals.SLEEP_MODIFIER` would help to set up a balance during too fast matching, otherwise the waiting timeout is already provided.

Reply:
Perhaps you are misunderstanding how `.exists()` works? `Globals.SLEEP_MODIFIER` would not effect the fastest match here. It only effects the acceptable slowest match, which is moot since the since the `.exists()`s are within a `while` loop and persists until a match is found. See [this page](http://doc.sikuli.org/region.html#Region.exists) for a description of the parameters.

Reply:
I was talking about `Util.kc_sleep` application after matching success to prevent fast clicking. And they are not quite the same - [Region.wait()](https://github.com/RaiMan/SikuliX1/blob/cfb86d3d35fb9cfe63f0ad65f0540677924860a0/API/src/main/java/org/sikuli/script/Region.java#L2231) also contains a while loop, [Region.exists()](https://github.com/RaiMan/SikuliX1/blob/cfb86d3d35fb9cfe63f0ad65f0540677924860a0/API/src/main/java/org/sikuli/script/Region.java#L2360) - not. This doesn't make sense at all, I just like `while region.exists(pattern, timeout)` more than, well...
```python
while not good and counter < treshold:
  try:
    region.wait(pattern, timeout)
    good = True
  except FindFailed:
    counter += 1
    if counter == treshold:
      raise RecoveryCallingException("msg")
```

Reply:
You really need to clarify what you're talking about when you jump from topic to topic. I obviously can't read your mind and your tendency to switch topics randomly is leading to a lot of confusion and needless back and forth.